go-controller/pkg/cluster/node.go

```diff
@@ -122,11 +130,54 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 		return nil
 	}
 
-	err = ovn.CreateManagementPort(node.Name, subnet.String(), clusterSubnets)
+	type readyFunc func(string) (bool, error)
+	var readyFuncs []readyFunc
+
+	macAddress, err := CreateManagementPort(node.Name, subnet.String(), clusterSubnets)
+	if err != nil {
+		return err
+	}
+
+	readyFuncs = append(readyFuncs, ManagementPortReady)
+	numFuncs := len(readyFuncs)
+	wg.Add(numFuncs)
+
+	// Set macAddress as annotation
+	err = cluster.Kube.SetAnnotationsOnNode(node, macAddress)
 	if err != nil {
+		logrus.Errorf("Failed to set node %s macAddress annotation: %v", node.Name, macAddress)
 		return err
 	}
 
+	portName := "k8s-" + node.Name
+
+	// Wait for the portMac to be created
+	for _, f := range readyFuncs {
+		go func(rf readyFunc) {
+			defer wg.Done()
+			if err := wait.PollImmediate(500*time.Millisecond, 2*time.Second, func() (bool, error) {
```

Comment:
@mccv1r0 We should kick this up to a much larger value than 2 seconds, like 3 or 5 minutes? It may take the master a long time to handle a bunch of nodes created at the same time.

Reply:
Good catch.  I made it lower when debugging.  I've set it to 300*time.Second which is what is used elsewhere in the node.go